src/Microsoft.AspNet.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -63,13 +65,12 @@ public class CollectionModelBinder<TElement> : IModelBinder
                     bindingContext,
                     valueProviderResult.RawValue,
                     valueProviderResult.Culture);
-                boundCollection = result.Model;
             }
 
-            model = bindingContext.Model;
+            var boundCollection = result.Model;
             if (model == null)
             {
-                model = GetModel(boundCollection);
+                model = GetModel(bindingContext.ModelType, boundCollection);
```

Comment:
Should this be named something more like `ConvertToCollectionType`? Or are there other uses?